File: TripKitAndroid/src/main/java/com/skedgo/tripkit/alerts/RealtimeAlertResponse.java
package com.skedgo.tripkit.alerts;

import com.google.gson.annotations.JsonAdapter;

import java.util.List;

import androidx.annotation.Nullable;

import static org.immutables.gson.Gson.TypeAdapters;
import static org.immutables.value.Value.Immutable;
import static org.immutables.value.Value.Style;

@Immutable
@TypeAdapters
@Style(passAnnotations = JsonAdapter.class)
@JsonAdapter(GsonAdaptersRealtimeAlertResponse.class)
public interface RealtimeAlertResponse {
    @Nullable
    List<AlertBlock> alerts();
}

File: TripKitAndroid/src/main/java/com/skedgo/tripkit/alerts/RealtimeAlertResponse.kt
package com.skedgo.tripkit.alerts

import com.google.gson.annotations.JsonAdapter
import org.immutables.gson.Gson.TypeAdapters
import org.immutables.value.Value.Immutable
import org.immutables.value.Value.Style

@Immutable
@TypeAdapters
@Style(passAnnotations = [JsonAdapter::class])
@JsonAdapter(
    GsonAdaptersRealtimeAlertResponse::class
)
interface RealtimeAlertResponse {
    fun alerts(): List<AlertBlock>
}